### ISSUE TITLE
super-linter internal validation breaks linting files starting with xx_bad_

### ISSUE THREAD

**Issue by cpeel** [github/super-linter#1288]
## Describe the bug

super-linter internal validation treats files that start with `xx_good_` and `xx_bad_` as special cases (see https://github.com/github/super-linter/blob/master/lib/functions/worker.sh#L102 and the files under https://github.com/github/super-linter/tree/master/.automation/test). This means that any file that starts with `xx_bad_` is assumed to fail validation. If it passes validation, the file is flagged as an error.

This breaks linting real files that happen to fit this pattern, for instance:

```
2021-03-01 21:32:16 [INFO]   File:[/github/workspace/tools/project_manager/handle_bad_page.php]
2021-03-01 21:32:16 [ERROR]   Found errors in [php] linter!
2021-03-01 21:32:16 [ERROR]   This file should have failed test case!
2021-03-01 21:32:16 [ERROR]   Error code: 0. Command output:
------
No syntax errors detected in /github/workspace/tools/project_manager/handle_bad_page.php
------
```

## Expected behavior

It's not reasonable to expect projects to use, or not use, any specific naming convention when using super-linter. If a file passes linting it should be reported as passing regardless of its filename.

## Steps to Reproduce

1. Create a valid PHP file called `handle_bad_page.php`
2. Lint the repo.
3. See linting error.

## Additional context

It seems reasonable to have super-linter have an environment variable used for running its internal tests and to only handle the special filename if that environment variable is provided, eg:

```diff
diff --git a/.github/workflows/stack-linter.yml b/.github/workflows/stack-linter.yml
index 7ce958e..25eb0cd 100644
--- a/.github/workflows/stack-linter.yml
+++ b/.github/workflows/stack-linter.yml
@@ -50,4 +50,5 @@ jobs:
           ACTIONS_RUNNER_DEBUG: true
           ERROR_ON_MISSING_EXEC_BIT: true
           VALIDATE_ALL_CODEBASE: false
+          SELF_TEST: true
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
diff --git a/lib/functions/worker.sh b/lib/functions/worker.sh
index 813e1cf..63d1a5b 100755
--- a/lib/functions/worker.sh
+++ b/lib/functions/worker.sh
@@ -95,11 +95,13 @@ function LintCodebase() {
       FILE_NAME=$(basename "${FILE}" 2>&1)
       DIR_NAME=$(dirname "${FILE}" 2>&1)

-      ############################
-      # Get the file pass status #
-      ############################
+      ###########################################
+      # Get the file pass status for self-tests #
+      ###########################################
       # Example: markdown_good_1.md -> good
-      FILE_STATUS=$(echo "${FILE_NAME}" | cut -f2 -d'_')
+      if [[ ! -z ${SELF_TEST} ]]; then
+        FILE_STATUS=$(echo "${FILE_NAME}" | cut -f2 -d'_')
+      fi

       ###################
       # Check if docker #
```

**Comment by ferrarimarco** [comment]
I think a solution _might be_ to apply this logic to files in the default test directory (tree). @admiralAwkbar ?

**Comment by admiralAwkbar** [comment]
@ferrarimarco i think this would work!
We already have the env var for tests so should just be a logic update

**Comment by cpeel** [comment]
Was this fixed or just closed?

**Comment by murrant** [comment]
Looks like just closed.  Just ran into it :)

https://github.com/librenms/librenms/runs/6141929420?check_suite_focus=true

**Comment by hepcat72** [comment]
Yep.  I ran into this issue *again*.  I think in #4027, I hadn't nailed it down to the file name including `xx_bad_...`, but that's definitely the case in https://github.com/PrincetonUniversity/ATACCompendium/pull/76
